lib/takeaway.rb

```diff
@@ -0,0 +1,44 @@
+require_relative 'menu'
+require_relative 'order'
+
+class Takeaway
+    attr_reader :menu, :order, :food_list
+    
+    def initialize(menu= Menu.new)
+        @menu = menu
+        @total_price = 0
+        @food_list = []
+        
+    end
+
+    def view
+        @menu.view
+    end
+
+    def pick(food, number)
+        new_order
+        number.times { food_list.push(food => @menu.menu[food])}
+        @food_list
+    end
+
+    def basket
+      @amount = @order.sum(@food_list) 
+      @amount
+      
+     end
```

Comment:
Don't create instance variables when not needed. Instance variables are only needed when state is shared between methods of an object. Here, this could be re-written as:
```rb
def basket
      @order.sum(@food_list) 
end
```

Also - basket is again not a very descriptive name. What is this method actually doing? Maybe `order_total` or `total_order_amount` would be better?